spyder/app/tour.py

```diff
@@ -103,6 +103,11 @@ def get_tour(index):
     qtconsole_link = "https://qtconsole.readthedocs.io/en/stable/index.html"
     #docs_link = __docs_url__
     docs_link = "https://docs.spyder-ide.org/current/index.html"
+    button_text = ""
+    if sys.platform != "darwin":
+        button_text = ("Please click on the button below to run some simple "
+                       "code in this console. This will be useful to show "
+                       "you other important features.")

```

Comment:
In order to have a common experience in all platforms, I think we should remove this button in all of them (instead of hiding it in macOS).

Reply:
Disabling the button also means the Console, Variable Explorer, History Log, etc. won't have anything in them to show users if the user hasn't already manually run code that session, which the tour currently refers to (since the bug appears to be triggered by the actual code execution, not the button press, so we can't just run the code automatically). Therefore, this is a significant downgrade in functionality and experience for the 80-90% of users not on Mac for the sake of a bug (possibly fixable in the future) that affects 10-20% that are.

There are a number of features we disable or enable on specific platforms due to bugs or otherwise (e.g. the fullscreen mode, which affects this very PR), and I'm not sure I really see the value in making the presence of the button consistent across platforms that would outweigh this, since the tour is self-contained and self-documenting with the text adapting according to whether the button is there, and perhaps unlike the regular UI, a user is very unlikely to run the same tour on multiple platforms to even notice it is different.

Reply:
Let's see what @juanis2112 thinks about this one. If she considers we should leave this button in Windows and Linux, then I'll defer to your decision.

My main concern is this could be confusing for students in classrooms if teachers tell them: "and now press the Run code button" because that button won't be available in macOS.

Reply:
> My main concern is this could be confusing for students in classrooms if teachers tell them: "and now press the Run code button" because that button won't be available in macOS.

I see, that's a good point. I'm not sure how likely it is that a teacher would have students simply step through the tour in class as opposed to showing them things themselves, having them complete tasks or do it on their own (since it doesn't require the user to actually do anything but click a few buttons; at least the professors I've known that teach with Spyder haven't). And in any case, students using Macs in classes where professors do that is a very small percentage of Spyder's userbase.

But if we want to avoid the chance of this confusion but not strip this feature from the vast majority of the Spyder userbase, instead of simply hiding the button we could disable it on macOS with a tooltip saying "Not available on macOS" or similar?

Reply:
> But if we want to avoid the chance of this confusion but not strip this feature from the vast majority of the Spyder userbase, instead of simply hiding the button we could disable it on macOS with a tooltip saying "Not available on macOS" or similar?

That's a pretty good compromise, thanks for the suggestion @CAM-Gerlach! I totally agree with it.

Reply:
Okay, sure. I'll make the appropriate suggestion below.